edb/lang/graphql/types.py

```diff
@@ -456,6 +418,20 @@ def __new__(mcls, name, bases, dct):
         return cls
 
 
+class CompleteShadowingFields:
```

Comment:
This class name is weird and deserves a docstring.

Reply:
I've thrown out that class and updated the code a bit to do without that extra complication.